### ISSUE TITLE
Implement real time exposure

### ISSUE THREAD

**Issue by tristan3214** [UWMRO/ScienceCamera#7]
Camera will run in "Run Till Abort" mode with the same general camera parameters as a single exposure.  Server thread will sleep and check for when the exposure is done and has been sent to the buffer loop.  Upon an image being done it will get written to the Linux temp folder where it will be erased when the Raspberry is shutdown.

In addition to the image being written, the file path will be queued up in a global variable.  The client will have their own timer that will request a file path from the server and display the image, which then recycles the process until the abort button is pressed.

There may be synchronizing issues between the two timers.  If the client is ahead of the server it will wait to start its timer again until a proper file path is inserted.  As for if the client lags, more thought needs to be put into this.

**Comment by tristan3214** [comment]
On the server, the camera will be set to run till abort and if the status is still acquiring and the thread successfully waited for the image to be acquired then the image is written to a temporary folder and the path is sent to the client.  If there is an abort then the server will cancel any waiting thread and not write to the temp folder.

Instead of the client requesting an image, they wait to be sent an image path at which time the progress bar will complete and then be restarted using the exposure time setting. 

Due to network latency and other such issues it is naive to attempt to have a progress bar that perfectly follows the acquisition time.  As such, user may see "hopping" in the progress bar.

**Comment by tristan3214** [comment]
Noticed an issue with aborting.  For some reason when a thread is being hanged by the Andor function WaitForAcquisition this somehow is blocking the whole server despite the server being multi-threaded with twisted's defer.deferToThread.  As a result during real time exposure, calls to get temperature are blocked until an acquisition is complete.  More importantly an abort call can not be completed until the acquisition is complete.  This is unacceptable behavior in that abort should be immediate.  This blocking also makes it so I can't call the Andor function CancelWait **during** the time the thread is being hanged.

As a counter example, with a single exposure the code simply waits by checking that the status is still acquiring; an abort can be easily called without any blocking in this case.  

With the nature that is run till abort the camera is always acquiring so the code needs another indicator that an acquisition has finished.  At this point all alternative attempts to create this other indicator has been fraught with failure due to lack of either my understanding of the drivers or a non-robust development of them.

**Comment by tristan3214** [comment]
Moving away from Andor's *WaitForAcquisition* function, with the indirect help of John P., I was able to implement *GetAcquisitionProgress* that iterates a long every time a new acquisition is done.

This may not be the best place to mention this, but when SWIG outputs a long I get nonsense in python (e.g. 139771120713728L comes back  for the number of acquisitions, in one instance).  The above was fixed by changing the following line of code in atmcdLXd.i:
`%apply float *OUTPUT {long *acc, long *series};`
changing float to **int** giving:
` %apply int *OUTPUT {long *acc, long *series};`

End of story is that when using long real time exposures abort has now an immediate effect.
